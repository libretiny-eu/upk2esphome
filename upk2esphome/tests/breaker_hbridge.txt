{
        "bt1_lv": 0,
        "bt1_pin": 17,
        "bt1_type": 0,
        "ch1_stat": 2,
        "ch_cddpid1": 9,
        "ch_dpid1": 1,
        "ch_num": 1,
        "chip_type": 3,
        "crc": 80,
        "ele_fun_en": 1,
        "jv": "1.0.1",
        "led1_lv": 0,
        "led1_pin": 9,
        "lose_vol": 76,
        "module": "CBU",
        "net_trig": 2,
        "netled1_lv": 0,
        "netled1_pin": 15,
        "netled_reuse": 0,
        "netn_led": 0,
        "nety_led": 1,
        "online_time": 10,
        "over_cur": 63000,
        "over_vol": 265,
        "reset_t": 5,
        "rl1_drvtime": 60,
        "rl1_type": 1,
        "rl_off1_lv": 1,
        "rl_off1_pin": 26,
        "rl_on1_lv": 1,
        "rl_on1_pin": 24,
        "temp_fun_en": 0,
        "vol_def": 0,
        "wide_vol": 0
}
