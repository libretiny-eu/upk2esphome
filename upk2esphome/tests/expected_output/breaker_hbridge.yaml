output:
  - platform: libretiny_pwm
    id: output_led_1
    pin:
      number: P9
      inverted: true

light:
  - platform: monochromatic
    id: light_switch_1
    output: output_led_1

binary_sensor:
  - platform: gpio
    id: binary_switch_1
    pin:
      number: P17
      inverted: true
      mode: INPUT_PULLUP
    on_press:
      then:
        - switch.toggle: switch_1

switch:
  - platform: hbridge
    id: switch_1
    name: Relay 1
    on_pin: P24
    off_pin: P26
    pulse_length: 60ms
    wait_time: 30ms
    on_turn_on:
      - light.turn_on: light_switch_1
    on_turn_off:
      - light.turn_off: light_switch_1

status_led:
  pin:
    number: P15
    inverted: true

uart:
  id: uart_bus
  tx_pin: TX1
  rx_pin: RX1
  baud_rate: 4800
  stop_bits: 1

sensor:
  - platform: bl0942
    uart_id: uart_bus
    current:
      name: BL0942 Current
    voltage:
      name: BL0942 Voltage
    power:
      name: BL0942 Power
      filters:
        multiply: -1
    energy:
      name: BL0942 Energy
    frequency:
      name: BL0942 Frequency
      accuracy_decimals: 2
